### PR TITLE
Test for fund campaign updater

### LINKED ISSUE #3048: Test: src/graphql/types/FundCampaign/updater.ts
# Goal
The goal of this issue is to improve the code coverage for the file: `src/graphql/types/FundCampaign/updater.ts`
# Tasks
1. Review the file to identify sections of code that are being ignored by Codecov or are not covered by tests.
2. Create or update test cases to ensure 100% coverage for the file.
3. Remove any `/* istanbul ignore */` or equivalent statements that bypass code coverage reporting, unless absolutely necessary.
# Resources
- Refer to the foundational documentation on writing test cases in the repository.
- Check the [Codecov report](https://app.codecov.io/gh/PalisadoesFoundation/talawa-api/tree/develop-postgres/src?displayType=list) for details on the uncovered lines.
# Acceptance Criteria
- All sections of the file are covered by tests.
- Code coverage for the file reaches 100%.
- PR created with necessary updates, passing all checks and reviews.

### Parent issue
https://github.com/PalisadoesFoundation/talawa-api/issues/2920

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Digvijay-x1**
…coverage for FundCampaign

**What kind of change does this PR introduce?** 
Tests

**Issue Number:**

Fixes #3048

**Snapshots/Videos:**
image

**If relevant, did you update the documentation?**

NO 

**Summary**

Added tests for the updater resolver in FundCampaign.

To do this, I had to export the updaterResolver from updater.ts so updater.test.ts could use it in tests.

I created the new updater.test.ts file and added 19  tests to get complete coverage.

The tests cover:

 - Auth: Not authenticated, user undefined, not an admin, no org memberships, non-admin org role.

  -  Data Validation: Fund doesn't exist, updater user doesn't exist (data corruption cases). 
  
  - fund: query fundsTable with currentUserId

  -  Logic: Null updaterId returns null, and the current user optimization check.

  -  Success: The main happy path for retrieving an existing updater.

**Does this PR introduce a breaking change?**
 NO  

### CodeRabbit AI Review
- [x] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [x] I have implemented or provided justification for each non-critical suggestion
- [x] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes 

## Summary by CodeRabbit

* **Refactor**
  * Improved backend resolver for campaign updater: stronger authentication & authorization checks, clearer error handling, and more reliable determination of the updater.

* **Tests**
  * Added comprehensive unit tests covering authentication, authorization, edge cases, and error scenarios to ensure correct updater resolution and logging.

### CODE REVIEW

**Comment by Digvijay-x1** [comment]
@palisadoes , requesting for your review and feedback , i am not able to find the reason why codecov is comparing with c0468a4 ? and failing . 
thanks .

**Comment by Digvijay-x1** [comment]
@coderabbitai please review

File: src/graphql/types/FundCampaign/updater.ts [hunk 1]
```diff
@@ -1,120 +1,127 @@
...
+
+	const currentUserId = ctx.currentClient.user.id;
+
+	const [currentUser, existingFund] = await Promise.all([
+		ctx.drizzleClient.query.usersTable.findFirst({
+			where: (fields, operators) => operators.eq(fields.id, currentUserId),
+		}),
+		ctx.drizzleClient.query.fundsTable.findFirst({
+			columns: {
+				isTaxDeductible: true,
```
Comment (palisadoes) [inline src/graphql/types/FundCampaign/updater.ts:31]: Why was this file edited?

Comment (Digvijay-x1) [reply src/graphql/types/FundCampaign/updater.ts:31]: I created an `updaterResolver` to export the resolver so that it can be imported in the test file, avoiding the need to rewrite the function there. These are the only two changes 
Screenshot From 2025-11-06 09-29-02
Screenshot From 2025-11-06 09-29-52

Comment (Digvijay-x1) [reply src/graphql/types/FundCampaign/updater.ts:31]: Also previously the `updater.ts` file was not linted properly which result in 4 linted changes . 
Screenshot 2025-11-06 132905

**Comment by palisadoes** [comment]
@coderabbitai full review

**Comment by palisadoes** [comment]
@coderabbitai approve the PR if everything is OK

**Comment by palisadoes** [comment]
@Digvijay-x1 Please ensure code rabbit approves your work

**Comment by Digvijay-x1** [comment]
@palisadoes, CodeRabbit is suggesting changes to the `updater.ts` file. I also have doubts about whether it should be `currentUserId` (original) or `parent.fundId` (which I thought was correct and raised as a bug fix in the previous closed PR #3700 ). However, you later corrected me that it should be currentUserId, so I wrote specific tests to ensure no one misinterprets this in the future.

**Comment by palisadoes** [comment]
> @palisadoes, CodeRabbit is suggesting changes to the `updater.ts` file. I also have doubts about whether it should be `currentUserId` (original) or `parent.fundId` (which I thought was correct and raised as a bug fix in the previous closed PR #3700 ). However, you later corrected me that it should be currentUserId, so I wrote specific tests to ensure no one misinterprets this in the future.

Please be clearer in your description of the issue and the logic of what the file under test does and how it relates to `currentUserId` and/or `parent.fundId`

**Review by palisadoes — APPROVED** [review]
